presto-main/src/main/java/io/prestosql/snapshot/SnapshotFileBasedClient.java

```diff
@@ -51,16 +53,24 @@ public SnapshotFileBasedClient(HetuFileSystemClient fsClient, Path rootPath)
     }
 
     @Override
-    public void storeState(SnapshotStateId snapshotStateId, Object state)
+    public void storeState(SnapshotStateId snapshotStateId, Object state, SnapshotDataCollector dataCollector)
```

Comment:
Should also capture the time to capture/processMarker and restore/resume methods for the time spend in serializing to memory also

Reply:
Modified